sdk/turing/router/config/common/schemas.py

```diff
@@ -0,0 +1,148 @@
+import re
+from abc import ABC, abstractmethod
+
+
+class Schema(ABC):
+    regex_exp = None
+
+    @classmethod
+    @abstractmethod
+    def verify_regex(cls, value):
+        pass
+
+
+class DockerImageSchema(Schema):
+    regex_exp = r"^([a-z0-9]+(?:[._-][a-z0-9]+)*(?::\d{2,5})?\/)?([a-z0-9]+(?:[._-][a-z0-9]+)*\/)*([a-z0-9]+(?:[._-][a-z0-9]+)*)(?::[a-z0-9]+(?:[._-][a-z0-9]+)*)?$"
```

Comment:
Conceptual question: Can we add all the regex into openapi spec itself (with `patten` keyword) and rely on openapi-generator to do validation for us?
Also, client side validation is good to have, but absolutely not essential to the SDK. It's different from the client-side validation on UI – the reason why we added it to UI is to validate form data dynamically, before user submitted request to the server (i.e. to prevent users from filling the entire form if they provided wrong input to one of the fields). This is motivation doesn't work for the SDK flow, as we will only validate the entire form after python script is executed.
So on the good side, client-side validation helps to report user-friendly error messages, but the drawback of it is that now we have to maintain validation schemas at three different places – API for server-side validation, UI and SDK – for client side validation. Would be nice to see if we can at least merge API and SDK validation into a single source of truth (i.e openapi spec itself)

Reply:
Apparently this is possible, according to the OpenAPI specs [here](https://swagger.io/docs/specification/data-models/data-types/):

> pattern
> The pattern keyword lets you define a regular expression template for the string value. Only the values that match this template will be accepted. The regular expression syntax used is from JavaScript (more specifically, ECMA 262). Regular expressions are case-sensitive, that is, [a-z] and [A-Z] are different expressions. For example, the following pattern matches a Social Security Number (SSN) in the 123-45-6789 format:

I don't think I've seen any issues with using OpenAPI Generator with regex patterns (at least with a quick search on Google) so it is possible to consider unifying all the regex patterns in the OpenAPI spec.

Reply:
As for the client-side validation, I was thinking maybe some users would use the SDK on notebooks so a quick exception thrown out while building the configs (cell by cell) would've been helpful but thanks for clarifying again about the primary usage of the SDK in the form of python script :)

Reply:
Ok it seems like the autogenerated classes validate the text values with the regex expressions correctly; I'll begin moving the schemas to the OpenAPI spec.